core-bundle/tests/Twig/Runtime/FigureRendererRuntimeTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Twig\Runtime;
+
+use Contao\CoreBundle\File\MetaData;
+use Contao\CoreBundle\Image\Studio\Figure;
+use Contao\CoreBundle\Image\Studio\FigureBuilder;
+use Contao\CoreBundle\Image\Studio\ImageResult;
+use Contao\CoreBundle\Image\Studio\Studio;
+use Contao\CoreBundle\Tests\TestCase;
+use Contao\CoreBundle\Twig\Runtime\FigureRendererRuntime;
+use Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException;
+use Twig\Environment;
+
+class FigureRendererRuntimeTest extends TestCase
+{
+    public function testConfiguresBuilder(): void
+    {
+        $metaData = new MetaData([]);
+
+        $configuration = [
+            'metaData' => $metaData,
+            'disableMetaData' => true,
+            'locale' => 'de',
+            'linkAttributes' => ['foo' => 'bar'],
+            'linkHref' => 'foo',
+            'lightBoxResourceOrUrl' => 'foobar',
+            'lightBoxSize' => '_lightbox_size',
+            'lightBoxGroupIdentifier' => '123',
+            'enableLightBox' => true,
+            'options' => ['foo' => 'bar'],
+        ];
+
+        $expectedFigureBuilderCalls = [
+            'from' => 'resource',
+            'setSize' => '_size',
+            'setMetaData' => $metaData,
+            'disableMetaData' => true,
+            'setLocale' => 'de',
+            'setLinkAttributes' => ['foo' => 'bar'],
+            'setLinkHref' => 'foo',
+            'setLightBoxResourceOrUrl' => 'foobar',
+            'setLightBoxSize' => '_lightbox_size',
+            'setLightBoxGroupIdentifier' => '123',
+            'enableLightBox' => true,
+            'setOptions' => ['foo' => 'bar'],
+        ];
+
+        $runtime = $this->getRuntime($expectedFigureBuilderCalls);
+
+        $this->assertSame('<result>', $runtime->render('resource', '_size', $configuration));
+    }
+
+    /**
+     * @testWith ["metaData", "setMetaData"]
+     */
+    public function testAllowsDefiningMetaDataAsArray(string $key): void
+    {
+        $metaData = [MetaData::VALUE_ALT => 'foo'];
+
+        $runtime = $this->getRuntime(['setMetaData' => new MetaData($metaData)]);
+
+        $this->assertSame('<result>', $runtime->render('resource', null, [$key => [MetaData::VALUE_ALT => 'foo']]));
+    }
+
+    public function testUsesCustomTemplate(): void
+    {
+        $runtime = $this->getRuntime([], '@App/custom_figure.html.twig');
+
+        $this->assertSame('<result>', $runtime->render(1, null, [], '@App/custom_figure.html.twig'));
+    }
+
+    public function testFailsWithInvalidConfiguration(): void
+    {
+        $this->expectException(NoSuchPropertyException::class);
```

Comment:
This will also expect exceptions in the `getRuntime()` method. Intended?

Reply:
Ah, yes you're right. Got me again.

Reply:
Fixed in a70beef2aba5099db37f8da41e03634608baf149